drivers/wifi/eswifi/eswifi_core.c

```diff
@@ -120,6 +120,27 @@ static void __parse_scan_res(char *str, struct wifi_scan_result *res)
 	}
 }
 
+char *eswifi_rsp_to_data(char *resp, size_t rlen)
+{
+	char *data = NULL;
+	int i;
+
+	/* format is \r\n[DATA]\r\nOK\r\n> or */
+	if (resp[0] != '\r' || resp[1] != '\n') {
+		return NULL;
+	}
+
+	for (i = 2; i < rlen; i++) {
```

Comment:
It's also a good idea to declare `i` as `size_t`, to avoid comparisons between signed and unsigned.  (Or overflows, should the sizeof(int) and sizeof(size_t) differ.)

Reply:
ok I'm going to fix that